libraries/AP_HAL_Linux/Scheduler.cpp

```diff
@@ -75,7 +75,7 @@ void Scheduler::init_realtime()
     mlockall(MCL_CURRENT|MCL_FUTURE);
 
     struct sched_param param = { .sched_priority = APM_LINUX_MAIN_PRIORITY };
-    if (sched_setscheduler(0, SCHED_FIFO, &param) == -1) {
+    if (pthread_setschedparam(pthread_self(), SCHED_FIFO, &param) == -1) {
```

Comment:
tsc tsc... musl decided not to implement the wrapper for a syscall because in Linux it doesn't conform with POSIX.
However since the pthread_setschedparam() is equivalent here I see no problem in us changing.  A slightly better commit description would be:  "musl doesn't provide sched_setscheduler() so switch to pthread_setschedparam() for compatibility".

Reply:
I did update the message with a more complete explanation.